app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/RecycleAdapter.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.demoappkotlin.listView_ViewPager.Adapters
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.CheckBox
+import android.widget.ImageView
+import android.widget.TextView
+import com.example.demoappkotlin.R
+import androidx.recyclerview.widget.RecyclerView
+import com.example.demoappkotlin.listView_ViewPager.ModelClasses.ModelRecycle
+
+class RecycleAdapter(var list: ArrayList<ModelRecycle>) :
+    RecyclerView.Adapter<RecycleAdapter.RecycleViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecycleViewHolder {
+        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_recycler_view, parent, false)
+        return RecycleViewHolder(itemView)
+    }
+
+    override fun onBindViewHolder(holder: RecycleViewHolder, position: Int) {
+        val listData = list[position]
+        holder.apply {
+            setIsRecyclable(false)
+            txtView?.text = listData.name
+            imgView?.setImageResource(listData.image)
+            chkbxView?.apply {
+                isChecked = listData.check
+                setOnCheckedChangeListener { btn, bool ->
+                    holder.chkbxView?.isChecked = bool
```

Comment:
`holder.chkbxView?.isChecked = bool` this line is necessary?  try to run without it and if required then utilize apply property.